packages/graph/hash_graph/lib/graph/src/store/mod.rs

```diff
@@ -379,6 +379,21 @@ pub trait EntityStore: for<'q> crud::Read<PersistedEntity, Query<'q> = Expressio
         entity_id: Option<EntityId>,
     ) -> Result<PersistedEntityIdentifier, InsertionError>;
 
+    /// Creates new entities.
+    ///
+    /// # Errors:
+    ///
+    /// - if the [`EntityType`] doesn't exist
+    /// - if the entities are not valid with respect to the specified [`EntityType`]
+    /// - if the account referred to by `owned_by_id` does not exist
+    /// - if an [`EntityId`] was supplied and already exists in the store
+    async fn create_entities(
+        &mut self,
+        entities: impl IntoIterator<Item = (Option<EntityId>, Entity), IntoIter: Send> + Send,
+        entity_type_id: VersionedUri,
```

Comment:
Is there a reason we're doing this for a single entity type? Why not just make that part of the iterator?

Is it to do with how we read from a stream during the insert? If so I think this should be captured in documentation in a **much** clearer and explicit way.

Reply:
Doing that for multiple entity types would require to read all entity type version ids for each entity, so it would be `N+2` statements instead of `3`

Reply:
I don't think I understand the issue here, what do you mean about `N+2` vs `3`

Reply:
If you want to insert `N` entities with
- different types: 
    1. Look up entity_type_version_id for *each* entity type
    2. Insert IDs
    3.  Insert entity
- same type: 
    1. Look up *one* entity_type_version_id
    2. Insert IDs
    3.  Insert entity

This could probably be optimized by first search for duplicated entity types before inserting the data, but this would need to traverse the iterators again. Also, it complicates the function a lot.

Reply:
And `entity_type_version_id` is an internal ID format to the DB, correct? It's not a versioned URI?

We should probably rename that column, it's very confusing.

Reply:
Yes. Additionally, if we do validation, later on, we need to query the type itself from the database as well and validate against each entity.

Reply:
Understood, when you move the code into bench, I'd like a doc comment explaining why we batch by entity type. And perhaps we should rename the function to be something more descriptive. `insert_entities_batched_by_type` or something.

Reply:
Considering we aren't moving it now, can we do the suggestion in place.

Reply:
Applied in 2148c6603e0b9931873ed5a898fc50ff299617c3

Reply:
Sorry can you also do this:
> I'd like a doc comment explaining why we batch by entity type

(Just because I know the first thing I'd do when setting up more benchmarks is to try and make this general over `(entity_type, entity_id, entity)`, and you could save a lot of time)

Reply:
Maybe it shouldn't be a doc comment on the trait, but instead just a comment above the relevant part of the code like
`// this would be a lot less efficient if we allowed for multiple entity types` or something

Reply:
I added this as a doc-comment as this will also show the documentation when hovering the function inside of an IDE:
<img width="598" alt="image" src="https://user-images.githubusercontent.com/21277928/193875116-061cbb09-3006-4627-becf-a44e0ff87476.png">